### ISSUE TITLE
ERROR valid-object-path

### ISSUE THREAD

**Issue by MaxiTalenti** [metabase/metabase#6362]
```
ERROR metabase.middleware :: GET /api/database 500 (7 ms) (3 DB calls)
{:message "Assert failed: (valid-object-path? %)",
:stacktrace
["models.permissions$object_path.invokeStatic(permissions.clj:87)"
"models.permissions$object_path.invoke(permissions.clj:87)"
"models.permissions$object_path.invokeStatic(permissions.clj:91)"
"models.permissions$object_path.invoke(permissions.clj:87)"
"models.table$perms_objects_set.invokeStatic(table.clj:48)"
"models.table$perms_objects_set.invoke(table.clj:47)"
"models.interface$fn__18406$G__18380__18413.invoke(interface.clj:93)"
"models.interface$make_perms_check_fn$_has_perms_QMARK___18439.invoke(interface.clj:140)"
"models.interface$fn__18389$G__18382__18398.invoke(interface.clj:93)"
"api.database$add_tables.invokeStatic(database.clj:46)"
"api.database$add_tables.invoke(database.clj:45)"
"api.database$dbs_list.invokeStatic(database.clj:140)"
"api.database$dbs_list.invoke(database.clj:138)"
"api.database$fn__33914$fn__33915.invoke(database.clj:149)"
"api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:248)"
"api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:243)"
"api.database$fn__33914.invokeStatic(database.clj:144)"
"api.database$fn__33914.invoke(database.clj:144)"
"middleware$enforce_authentication$fn__29528.invoke(middleware.clj:122)"
"api.routes$fn__43818.invokeStatic(routes.clj:60)"
"api.routes$fn__43818.invoke(routes.clj:60)"
"routes$fn__44498$fn__44499.doInvoke(routes.clj:75)"
"routes$fn__44498.invokeStatic(routes.clj:71)"
"routes$fn__44498.invoke(routes.clj:71)"
"middleware$log_api_call$fn__29627$fn__29629.invoke(middleware.clj:330)"
"middleware$log_api_call$fn__29627.invoke(middleware.clj:329)"
"middleware$add_security_headers$fn__29577.invoke(middleware.clj:245)"
"middleware$bind_current_user$fn__29532.invoke(middleware.clj:142)"
"middleware$maybe_set_site_url$fn__29581.invoke(middleware.clj:268)"]}
```

- Your operating system: Windows 7
- Your databases: SQL Server
- Metabase version: 0.26.2
- Metabase hosting environment: Windows 7

**When i try to write a query, i can´t write te query, no appear the model database, etc.
I try deleting the database but the error persist**

![metabase1](https://user-images.githubusercontent.com/7159779/32660915-96d0e640-c603-11e7-886f-fd4e89257313.jpg)
![metabase2](https://user-images.githubusercontent.com/7159779/32660916-9705dfa8-c603-11e7-806d-0ff0c518a8ae.jpg)
![metabase3](https://user-images.githubusercontent.com/7159779/32660917-9745c492-c603-11e7-80d1-2f8f6ee9b8af.jpg)
![metabase4](https://user-images.githubusercontent.com/7159779/32660918-978386f6-c603-11e7-8cdf-b7af8559a012.jpg)

**Comment by nadilas** [comment]
Same here, any progress / solution?

**Comment by nadilas** [comment]
+1

**Comment by Siddarth434** [comment]
Clear your browser cookies. It worked for me after clearing cookies and cache.

**Comment by nadilas** [comment]
@Siddarth434 okay, so I presume you mean after the database sync is ran. This doesn't help. I somehow have ` :error (not ("Valid permissions object path." a-java.lang.String))` this as an error show up on the screen.
I can see in the log that all tables and fields are caught and indexed properly. It just doesn't show up in the UI.

**Comment by MaxiTalenti** [comment]
Any progress / solution?

**Comment by nadilas** [comment]
@maxitalenti the only workaround I found Was to add permission for the connecting user to all schemas...

**Comment by lorn** [comment]
@nadilas I guess I have the same problem on #8693 

So the use that connects to the database, need permission to see all the schemas?

**Comment by Siddarth434** [comment]
@iorn. Yes. But not to all schemas. You should have access to schemas you are trying to add to metabase. And if metabase is hosted on different system or server, both the systems should have access to each other or Sql port should be open for both to communicate.

**Comment by nadilas** [comment]
@siddarth434 In my case this is not correct. I probably missed a step because for the life of me I couldn’t select which schemas I would like to add, simply nothing was showing up in the GUI. As soon as the connecting user had access to all schemas within the database, the problems (error message in console and GUI) went away. 

@iorn so yes, this is happened in my case, also with an mssql database. Your error message looks way too familiar to me :)

**Comment by 206siva** [comment]
Hive integrate with metabase through prestoDB

Every Thing is going good but when i add a database i am getting this error. what should i need to do for clear these error…

metabase.util.i18n.UserLocalizedString cannot be cast to java.lang.String

GET /api/user/current 401 (151 µs) (0 DB calls).

**Comment by flamber** [comment]
Closing, since it looks like it was fixed in 0.31
